state-chain/runtime/src/chainflip.rs

```diff
@@ -620,10 +621,30 @@ impl ChainEnvironment<Vec<SolAddress>, AltConsensusResult<Vec<SolAddressLookupTa
 	for SolEnvironment
 {
 	fn lookup(
-		_alts: Vec<SolAddress>,
+		alts: Vec<SolAddress>,
 	) -> Option<AltConsensusResult<Vec<SolAddressLookupTableAccount>>> {
-		// TODO: To be implemented with ALT election refactor
-		Some(AltConsensusResult::ValidConsensusAlts(vec![]))
+		let mut res = vec![];
+
+		// Verify all the data is ready
+		for alt in &alts {
+			if let Ok(Some(consensus)) =
+				solana_elections::SolanaAltWitnessingElectoralAccess::unsynchronised_state_map(alt)
```

Comment:
I would be nice to not have to deal with elections pallet internals here. 

For example the `ExactValue` election could expose an api like this:

```
// This method already exists:
fn witness_exact_value(id: ES::Identifier);
// It should be possible to implement this, where ES::Result is the AltConsensusResult
fn check_value(id: ES::Identifier) -> ES::Result;
```